### PR TITLE
[s] Fixes Mechs Protecting Their Pilots From the Crushing Force of a 2 000 Metric Ton Shuttlecraft.

### PR DESCRIPTION
**Pull request by Fordoxia**
## What Does This PR Do
Fixes mechs allowing their pilots to survive being shuttlegibbed.

## Why It's Good For The Game
Prevents the use of mechs for hijack cheese. Also it just doesn't make sense.

## Testing
Stood under shuttle.
Died.

## Changelog
:cl:
fix: The shuttle actually kills you if you're a mech pilot, now.
/:cl:

### CODE REVIEW

File: code/modules/shuttle/shuttle.dm [hunk 1]
```diff
@@ -640,6 +640,13 @@
 					var/obj/machinery/atmospherics/supermatter_crystal/bakoom = AM
 					addtimer(CALLBACK(bakoom, TYPE_PROC_REF(/obj/machinery/atmospherics/supermatter_crystal, explode), bakoom.combined_gas, bakoom.power, bakoom.gasmix_power_ratio), 1 SECONDS)
 				continue
+			// Your mech will not save you.
+			if(ismecha(AM))
+				var/obj/mecha/mech = AM
+				if(mech.occupant)
+					var/mob/pilot = mech.occupant
+					mech.Destroy()
+					pilot.gib()
```
Comment (BiancaWilkson) [inline code/modules/shuttle/shuttle.dm:649]: You'll need these to be wrapped in an INVOKE_ASYNC call to make this pass CL

Comment (SteelSlayer) [reply code/modules/shuttle/shuttle.dm:649]: Also, never call Destroy directly. Always use `qdel`

**Review by SteelSlayer — APPROVED** [review]

File: code/game/mecha/mecha.dm [hunk 2]
```diff
@@ -1317,6 +1317,20 @@
...
 	go_out()
 
+// Called when a shuttle lands on top of the mech. Prevents the mech from just dropping the pilot unharmed inside the shuttle when destroyed.
+/obj/mecha/proc/get_out_and_die()
+	var/mob/living/pilot = occupant
+	pilot.visible_message(
+		"<span class='warning'>[src] is hit by \
+		a hyperspace ripple!</span>",
+		"<span class='userdanger'>You feel an immense \
+		crushing pressure as the space around you ripples.</span>"
```
Comment (Burzah) [inline code/game/mecha/mecha.dm:1327]:
```suggestion
		"<span class='warning'>[src] is hit by a hyperspace ripple!</span>",
		"<span class='userdanger'>You feel an immense crushing pressure as the space around you ripples.</span>"
```
These should be on the same line. The lengths here do not warrant breaking them up.

Comment (Fordoxia) [reply code/game/mecha/mecha.dm]: Done.

**Review by DGamerL — APPROVED** [review]

**Pull request merged**